cloud/pkg/synccontroller/objectsync.go

```diff
@@ -44,8 +44,13 @@ func (sctl *SyncController) manageObject(sync *v1alpha1.ObjectSync) {
 		newObject.SetNamespace(sync.Namespace)
 		newObject.SetName(sync.Spec.ObjectName)
 		newObject.SetUID(types.UID(getObjectUID(sync.Name)))
-		msg := buildEdgeControllerMessage(nodeName, sync.Namespace, resourceType, sync.Spec.ObjectName, model.DeleteOperation, newObject)
-		beehiveContext.Send(commonconst.DefaultContextSendModuleName, *msg)
+		if msg := buildEdgeControllerMessage(nodeName, sync.Namespace, resourceType, sync.Spec.ObjectName, model.DeleteOperation, newObject); msg != nil {
```

Comment:
Will the msg be nil? Which kind of message?

Reply:
if objectsync's spec.ObjectKind is empty, then buildEdgeControllerMessage would return nil, such as pvc's objectsync

Reply:
Can you explain why the ObjectKind of pvc's objectsync is empty?

Reply:
1. edged query a pvc
2. edgecontroller upstream receive the msg
3. edgecontroller upstream get pvc from k8s api 
4. edgecontroller store pvc object to msg.content, and send msg to cloudhub
5. cloudhub send msg to edgehub
6. cloudhub get response from edge, and create pvc's objectsync
7. objectsync.spec.objectkind = util.GetMessageResourceType(msg)

step 7 msg and step 4 msg are same one, and msg.content is pvc object, which metatype is empty. So we need SetMetaType

Reply:
OK, get it, thanks! And if the msg is empty here, is it reasonable to delete the objectsync as below?

Reply:
And if we add the `util.SetMetaType(obj.(runtime.Object))`, maybe the msg will not be empty? :)

Reply:
there are three cases:
1. objectsync's resource is deleted
2. objectsync's resource is deleted, but a same name resource created
3. objectsync.Spec.ObjectKind is empty.

case 1: program will entry line 48, send delete msg to edge and then delete objectsync
case 2: program will entry line 73, sendEvents function will send delete msg to edge and then delete objectsync
case 3: program will entry line 50, delete objectsync directly

maybe msg will not be empty, add a check just for comfort. ：）

Reply:
Make sense, thanks for the clarification!